### PR TITLE
[#442] account for lists in `YamlUtils#loadYaml`

### PR DESCRIPTION
**Pull request by ewilkins-csi**
Also fixes a bug with the `hasDouble`, `hasInt`, and `hasBoolean` methods of `YamlObject`. Adds tests for `YamlUtils` including one to catch the root list issue. Finally, also updates the POM migration to account for POMs without dependency management sections.

### CODE REVIEW

File: foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/util/YamlUtils.java [hunk 1]
```diff
@@ -78,23 +97,23 @@ public String getString(String... path) {
         }
 
         public boolean hasInt(String... path) {
-            return hasValue(int.class, path);
+            return hasValue(Integer.class, path);
```
Comment (ewilkins-csi) [inline foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/util/YamlUtils.java:100]: I: these were returning false when they shouldn't have as the map has the boxed version of the value so `isInstance` was failing.

File: foundation/foundation-upgrade/src/main/resources/migrations.json [hunk 2]
```diff
@@ -73,7 +73,8 @@
         "implementation": "com.boozallen.aissemble.upgrade.migration.v1_10_0.DeltaSparkYamlMigration",
         "fileSets": [
           {
-            "includes": ["**/*.yaml"]
+            "includes": ["**/*.yaml"],
+            "excludes": ["**/target/**", "**/templates/**"]
```
Comment (ewilkins-csi) [inline foundation/foundation-upgrade/src/main/resources/migrations.json:77]: I: Not strictly necessary but this cuts down significantly on the number of output lines talking about un-parseable helm templates.

File: foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/utils/YamlUtilsSteps.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        String type = yaml.get(name).getClass().getSimpleName();
+        assertTrue("Property was not loaded as string: " + name + " (was " + type + ")", yaml.hasString(name));
+        assertEquals("Property value does not match for: " + name, value, yaml.getString(name));
+    }
+
+    @Then("the decimal value of the property {string} is {double}")
+    public void theDecimalValueOfThePropertyIs(String name, double value) {
+        String type = yaml.get(name).getClass().getSimpleName();
+        assertTrue("Property was not loaded as double: " + name + " (was " + type + ")", yaml.hasDouble(name));
+        assertEquals("Property value does not match for: " + name, value, yaml.getDouble(name), 0.01);
```
Comment (csun-cpointe) [inline foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/utils/YamlUtilsSteps.java:53]: Q: `0.01`?

**Review by csun-cpointe — APPROVED** [review]

File: foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/utils/YamlUtilsSteps.java (hunk 3 above)
Comment (ewilkins-csi) [reply foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/utils/YamlUtilsSteps.java:53]: It's a `delta` param that they added to floating point assertions. Avoids the tricky floating point rounding issues where you get `0.000000000001` or some other minuscule difference just from how FP is represented. So this says treat the numbers as equal if they are within `0.01` of each other.

**Review by carter-cundiff — APPROVED** [review]

**Pull request merged**